src/frontend/components/UI/QuestsViewer/components/QuestDetailsWrapper/index.tsx

```diff
@@ -89,6 +93,63 @@ async function getRewardClaimGasEstimation(reward: Reward) {
   return gasNeeded
 }
 
+interface ConfirmProps extends ModalProps {
+  networkName: string
+  onConfirm: () => void
+  onCancel: () => void
+}
+
+function ConfirmClaimModal(props: ConfirmProps) {
+  const { t } = useTranslation()
```

Comment:
should we move this to a separate components folder?

Reply:
I don't think so, this component will only be used here so I think it is better to keep them close.

Reply:
this still goes against the pattern we use elsewhere. If this component is the only one that uses it, we would normally just put it in a subfolder alongside this component

Reply:
true!